client/src/components/FriendCard.js

```diff
@@ -1,72 +1,106 @@
+		display: 'flex',
+		width: '100%',
+		zIndex: 1
+	},
+	cardMedia: {
+		width: 160
+	},
+	cardContent: {
+		width: '100%',
+		marginTop: theme.spacing(2)
+	},
+	large: {
+		width: theme.spacing(8),
+		height: theme.spacing(8),
+		margin: theme.spacing(1),
+		zIndex: 1,
+		alignItems: 'left'
+	},
+	box: {
+		zIndex: 0,
+		backgroundColor: '#FF7F50',
+		opacity: 0.4
+	}
 }));
 
-export default function ProductCard(friend) {
-    const classes = useStyles();
+export default function FriendCard(props) {
+	const classes = useStyles();
+	const authContext = useContext(AuthContext);
 
-    return (
-        <CardActionArea>
-        <Box position="relative" left="0px">
-        <Card className={classes.card} >
-            <Box width="100%" height="28%" className={classes.box} position="absolute" left="0%" borderRadius="borderRadius"/>
-            {/* <CardMedia className={classes.cardMedia} image={friend.image} title={friend.name}/> */}
-            <Box  position="relative" left="0%" alignItems='left'>
-            <Avatar alt={friend.name} src={friend.image} className={classes.large}/>
-            </Box>
-            <div className={classes.cardDetails}>
-                <CardContent className={classes.cardContent}>
-                    <Typography component="h2" variant="h5">
-                        {friend.name}
-                    </Typography>
-                    {/* <Typography variant="subtitle1" paragraph>
-                        {friend.description}
-                    </Typography> */}
-                </CardContent>
-                <CardActions>
-                    <Button variant="outlined" color="primary" href="#" size="small">follow</Button>
-                </CardActions>
-            </div>
-        </Card>
-        </Box>
-        </CardActionArea>
-    );
-}
\ No newline at end of file
+	const followUser = async (id) => {
+		try {
+			const res = await axios.post(`/users/follow/${id}`);
+			authContext.handleFollow(id);
+			return res;
+		} catch (error) {
+			console.log('error following user');
+		}
+	};
+	const unfollowUser = async (id) => {
+		try {
+			const res = await axios.post(`/users/unfollow/${id}`);
+			authContext.handleUnfollow(id);
+			return res;
+		} catch (error) {
+			console.log('error unfollowing user');
+		}
+	};
+	return (
+		<CardActionArea>
+			<Box position="relative" left="0px">
+				<Card className={classes.card}>
+					<Box
+						width="100%"
+						height="28%"
+						className={classes.box}
+						position="absolute"
+						left="0%"
+						borderRadius="borderRadius"
+					/>
+					<Box position="relative" left="0%" alignItems="left">
+						<Avatar alt={props.name} src={props.profile_picture} className={classes.large} />
+					</Box>
+					<div className={classes.cardDetails}>
+						<CardContent className={classes.cardContent}>
+							<Typography component="h2">{props.name}</Typography>
+						</CardContent>
+						<CardActions>
+							{authContext.friends_list.includes(props.id) ? (
```

Comment:
similarly the friendList could be an object too. These are optimizations maybe we can make later

Reply:
Yeah as to both of those notes, The nested ternaries and chained functions do make me a little nervous haha. Would a better solution eventually be to abstract this logic away to context & create more subcomponents?

Reply:
That's definitely a solution yeah